### PR TITLE
Issue #7651: add examples for NestedForDepth

### PR DESCRIPTION
**Pull request by romani**
Issue #7651

image

```
$ cat config.xml 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
    "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
    "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="NestedForDepth"/>
    </module>
</module>
$ cat Test.java 
class Test {
 public void foo() {
    for(int i=0; i<10; i++) {
        for(int j=0; j<i; j++) {
            for(int k=0; k<j; k++) { // violation
            }
        }
    }

    for(int i=0; i<10; i++) {
        for(int j=0; j<i; j++) { // ok
        }
    }

 }
}
$ java $RUN_LOCALE -jar /var/tmp/checkstyle-8.33-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:5:13: Nested for depth is 2 (max allowed is 1). [NestedForDepth]
Audit done.
Checkstyle ends with 1 errors.

 
$ cat config.xml 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
    "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
    "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="NestedForDepth">
          <property name="max" value="2"/>
        </module>
    </module>
</module>
$ cat Test.java 
class Test {
 public void foo() {
    for(int i=0; i<10; i++) {
        for(int j=0; j<i; j++) {
            for(int k=0; k<j; k++) {
                for(int l=0; l<k; l++) { // violation
                }
            }
        }
    }

    for(int i=0; i<10; i++) {
        for(int j=0; j<i; j++) {
            for(int k=0; k<j; k++) { // ok
            }
        }
    }

 }
}
$ java $RUN_LOCALE -jar /var/tmp/checkstyle-8.33-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:6:17: Nested for depth is 3 (max allowed is 2). [NestedForDepth]
Audit done.
Checkstyle ends with 1 errors.
```

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NestedForDepthCheck.java [hunk 1]
```diff
@@ -41,14 +41,46 @@
...
- * To configure the check to allow nesting depth 3:
+ * To configure the check to allow nesting depth 2:
  * </p>
  * <pre>
  * &lt;module name=&quot;NestedForDepth&quot;&gt;
- *   &lt;property name=&quot;max&quot; value=&quot;3&quot;/&gt;
+ *   &lt;property name=&quot;max&quot; value=&quot;2&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
+ * <p>Example: </p>
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NestedForDepthCheck.java:66]: no need for extra space, just `<p>Example:</p>` is fine

File: src/xdocs/config_coding.xml [hunk 2]
```diff
@@ -4028,15 +4028,46 @@ for (String line: lines) {
         <source>
 &lt;module name=&quot;NestedForDepth&quot;/&gt;
         </source>
+        <p>Example: </p>
```
Comment (strkkk) [inline src/xdocs/config_coding.xml:4031]: same here

File: src/xdocs/config_coding.xml [hunk 3]
```diff
@@ -4028,15 +4028,46 @@ for (String line: lines) {
...
+        <source>
+for(int i=0; i&lt;10; i++) {
+    for(int j=0; j&lt;i; j++) {
+        for(int k=0; k&lt;j; k++) { // violation
```
Comment (strkkk) [inline src/xdocs/config_coding.xml:4035]: `// violation` -> `// violation, max allowed nested loop number is 1`

**Review by strkkk — CHANGES_REQUESTED** [review]

File: src/xdocs/config_coding.xml (hunk 3 above)
Comment (romani) [reply src/xdocs/config_coding.xml:4035]: fixed

File: src/xdocs/config_coding.xml (hunk 2 above)
Comment (romani) [reply src/xdocs/config_coding.xml]: fixed.

**Review by strkkk — APPROVED** [review]

**Review by pbludov — APPROVED** [review]

**Pull request merged**